### ISSUE TITLE
Syncthingy GUI doesn't pull up on GNOME Desktop.

### ISSUE THREAD

**Issue by TutorialsOnDeck** [zocker-160/SyncThingy#37]
Tested with Fedora Linux 41.20241120.0 (Silverblue) - Fresh install and default user settings out of the box.
Wayland compositor.
Syncthingy v0.75
Syncthing v1.28.0

With a completely Vanilla GNOME Desktop:
The Syncthingy program does not create a GUI menu either from clicking the icon in Apps or finding the already running program under the menu in the upper right (shows as "1 Background App").

Syncthingy is shown running as a 'background App' and syncthingy + syncthing are shown in HTOP / task manager. All functions are working except the GUI for Syncthingy never appears.

If you use plugins to alter your UI, such as [Appindicator](https://extensions.gnome.org/extension/615/appindicator-support/), Clicking the Syncthingy icon generated by this plugin on the panel will reveal the "choose your option" menu and allows the Settings GUI to appear as expected.

If I had to guess why this occurs, it is because Syncthingy does not generate an app window / GUI until you select "Settings". Gnome may not recognize popup menus as actual Apps and thus no App GUI is generated. 

GNOME must be either blocking/suppressing the selection menu from popping up or is not coded to acknowledge such menus from Background App indicators. A workaround may be to have the Syncthingy application spawned / generated on click instead of providing a "what do you want to do" menu. Do away with the menu and just add a WebUI config link and all other features to the Syncthingy 'Settings' popup instead.

It sucks having to potentially rewrite YOUR program to fit what GNOME thinks is the correct way of doing things but GNOME is the flagship DE for many corporate Linux distros. I am not telling you what you should do with your program; this just popped up in my deployment testing. I use KDE personally.

**Comment by zocker-160** [comment]
Thank your for this report, sadly you are not the first reporting issues with GNOME so I am aware of it.

First of all, I personally am testing this application on Ubuntu GNOME and there the tray icon + menu do work fine with the limitation that left-click and right-click are both registered as "request context menu" (unlike in KDE).

I suspect that Ubuntu has changed their version of GNOME is a way to make this work, I think you can understand that I cannot test all of the possible Distro + DE + potential patches combinations available out there.

> GNOME must be either blocking/suppressing the selection menu from popping up or is not coded to acknowledge such menus from Background App indicators.

Is there a "proper" way of implementing tray icons or if this is just something GNOME developers don't want to exist. 
Sadly there is not much I can do here anyway, since all I am doing is using QT's `QSystemTrayIcon` implementation.

> A workaround may be to have the Syncthingy application spawned / generated on click instead of providing a "what do you want to do" menu.

I do like this idea in general, but there are sadly a few problems that would make usage more complicated from a user's perspective.
Right now left-click (mouse primary) opens the WebUI and right-click (mouse secondary) opens the context menu. IMO this is intuitive and expected behavior. 
The reason why I don't spawn a window when starting the application is to make it easy for users to add it to autostart without having to click a window away every time or having to provide additional CLI arguments. 

Originally this application was made for the Steam Deck where making it as simple as possible way my primary goal.

TL;DR:  
I am willing to implement workarounds to make it work on GNOME, but currently I am not aware of a "proper" way of implementing tray indicators on vanilla GNOME given that (from what I can see) many other applications suffer from the same problem.

**Comment by zocker-160** [comment]
Update from my side:

I just looked into the source code and I completely forgot, that I am already implementing what you are proposing. Currently I am opening a window and resizing it to (0, 0) to make the tray icon even show up at all.

That this doesn't seem to work anymore is unfortunate, but sadly nothing really I can do.

**Comment by zocker-160** [comment]
I just tested on Ubuntu 24.04 and it works fine
![grafik](https://github.com/user-attachments/assets/c3e81f81-bf61-4442-a1cb-43939df8dd29)

I don't think here is anything actionable for me.

**Comment by dtfleetwood** [comment]
Just installed it on Fedora Silverblue 41 and I'm not having any issues.

**Comment by zocker-160** [comment]
You can only run a single instance at the same time.

**Comment by zeronacer** [comment]
I'm sorry that I didn't notice that.

I killed every process via the System Monitor, but the tray icon still doesn't show up. Process itself seems to be running though, and the health endpoint has a positive response.

```
❯ flatpak run com.github.zocker_160.SyncThingy
QSocketNotifier: Can only be used with threads started with QThread
qt.qpa.qgnomeplatform: Could not find color scheme  ""
config watcher running
running inside Flatpak 

Using Icon: "com.github.zocker_160.SyncThingy"
QObject::connect: No such signal QPlatformNativeInterface::systemTrayWindowChanged(QScreen*)
Ping endpoint:  "http://127.0.0.1:8384/rest/noauth/health"
Requesting background permission...
ERROR:  QNetworkReply::ConnectionRefusedError
reply:  QJsonDocument()
HTTP check failed, retry with HTTPS
Ping endpoint:  "https://127.0.0.1:8384/rest/noauth/health"
reply:  QJsonDocument()
Syncthing does not seem to be running, starting own instance
/app/bin/syncthing
Background / Autostart permission granted
```

**Comment by zocker-160** [comment]
Do tray icons of other applications work?

Maybe this line
`QObject::connect: No such signal QPlatformNativeInterface::systemTrayWindowChanged(QScreen*)`

might be a hint and looking online there seem to be others with a similar issue.

Sadly I don't think there is anything I can do here from my side.

**Comment by zeronacer** [comment]
Other tray icons work as expected. If the error doesn't ring any immediate bells for you, I'll search online and refer back here if I find a solution. 

Thanks for your time!

**Comment by zeronacer** [comment]
I switched from the "Tray Icons: Reloaded" extension to the (native?) "AppIndicator and KStatusNotifierItem Support" extension, and it has worked like a charm ever since. I had the former left over from upgrading my linux installation.

**Comment by LunarEclipse363** [comment]
Hi, newer versions of GNOME without any third party extensions use [a background activity portal](https://flatpak.github.io/xdg-desktop-portal/docs/doc-org.freedesktop.portal.Background.html) for this, I'm not sure how all the functionality maps but perhaps it's possible to implement for better system integration.

The portal also seems to facilitate autostart for Flatpak apps.

Edit: there is [libportal](https://github.com/flatpak/libportal) which seems to come with GTK3/4 and Qt5/6 integration so perhaps that would be a route for this.

**Comment by zocker-160** [comment]
@LunarEclipse363 I am already using the background portal.
